### PR TITLE
Resolution settings referenced from DB record for 3dsMax

### PR DESCRIPTION
**Pull request by moonyuet**
## Changelog Description
1. Add Callback for setting the resolution according to DB after the new scene is created.
2. Add a new Action into openpype menu which allows the user to reset the resolution in 3dsMax

image

## Additional info
N/A

## Testing notes:
1. launch 3dsMax via Openpype
2. check the resolution in render setting
4. it should be set according to DB.
Reset Resolution
1. locate "Set Resolution" in openpype menu
2. click it
3. check the resolution in render setting to see if it is same with the setting shown in DB

### CODE REVIEW

**Review by LiborBatek — DISMISSED** [review]
Whem opening the new shot for first time, OP correctly sets the ```Image Resolution```.

When triggering the ```Set Resolution``` action the resolution gets correct values (if they differ before).

All working as expected!

File: openpype/hosts/max/api/lib.py [hunk 1]
```diff
@@ -157,6 +164,63 @@ def get_multipass_setting(project_setting=None):
...
+    # Set project resolution
+    width_key = "resolutionWidth"
+    height_key = "resolutionHeight"
+    proj_width_key = project_data.get(width_key, 1920)
+    proj_height_key = project_data.get(height_key, 1080)
+
+    width = asset_data.get(width_key, proj_width_key)
+    height = asset_data.get(height_key, proj_height_key)
+
+    set_scene_resolution(width, height)
```
Comment (antirotor) [inline openpype/hosts/max/api/lib.py:205]: We should stop using `legacy_io`. Better to use `get_current_project` and `get_current_project_name()` from `openpype.pipeline`:

```suggestion
    project_resolution_data = get_current_project(fields=["data.resolutionWidth", "data.resolutionHeight"])["data"]
    asset_resolution_data = get_current_project_asset(fields=["data.resolutionWidth", "data.resolutionHeight"])["data"]

    # Set project resolution
    project_width = int(project_resolution_data.get("resolutionWidth", 1920))
    project_height = int(project_resolution_data.get("resolutionHeight", 1080))

    width = int(asset_resolution_data .get("resolutionWidth", project_width))
    height = int(asset_resolution_data .get("resolutionHeight", project_height))

    set_scene_resolution(width, height)
```

File: openpype/hosts/max/api/lib.py [hunk 2]
```diff
@@ -157,6 +164,63 @@ def get_multipass_setting(project_setting=None):
                            ["multipass"])
 
 
+def set_scene_resolution(width, height):
```
Comment (antirotor) [inline openpype/hosts/max/api/lib.py:167]:
```suggestion
def set_scene_resolution(width: int, height: int):
```

File: openpype/hosts/max/api/lib.py [hunk 3]
```diff
@@ -6,6 +6,13 @@
 from typing import Union
 import contextlib
 
+from openpype.client import (
+    get_project
+)
+from openpype.pipeline import legacy_io
+
+from openpype.pipeline.context_tools import get_current_project_asset
```
Comment (antirotor) [inline openpype/hosts/max/api/lib.py:14]:
```suggestion
from openpype.pipeline.context_tools import get_current_project_asset, get_current_project
```

**Review by antirotor — CHANGES_REQUESTED** [review]
Just a quick note about use of `legacy_io`. Suggestions are not tested.

**Review by antirotor — APPROVED** [review]
Looks good now

**Comment by LiborBatek** [comment]
Doenst work for me now...unfortunatelly: get this error and the resolution is not changed acccording to DB
image

**Comment by moonyuet** [comment]
> Doenst work for me now...unfortunatelly: get this error and the resolution is not changed acccording to DB image

it is mainly because I have added extra ["data"] in my code, Will update the code tmr.

File: openpype/hosts/max/api/lib.py [hunk 4]
```diff
@@ -157,6 +162,60 @@ def get_multipass_setting(project_setting=None):
...
+    any information regarding scene resolution .
+
+    Returns:
+        None
+    """
+    data = ["data.resolutionWidth", "data.resolutionHeight"]
+    project_resolution = get_current_project(fields=data)["data"]
+    project_resolution_data = project_resolution["data"]
+    asset_resolution = get_current_project_asset(fields=data)["data"]
+    asset_resolution_data = asset_resolution["data"]
```
Comment (BigRoy) [inline openpype/hosts/max/api/lib.py:193]:
```suggestion
    project_resolution = get_current_project(fields=data)
    project_resolution_data = project_resolution["data"]
    asset_resolution = get_current_project_asset(fields=data)
    asset_resolution_data = asset_resolution["data"]
```

Comment (moonyuet) [reply openpype/hosts/max/api/lib.py]: Thanks, updated the PR for the removal of the attributes

**Review by antirotor — APPROVED** [review]

**Pull request merged**